gis/openstreetmap-carto/Portfile

```diff
@@ -0,0 +1,116 @@
+
+description         A general-purpose OpenStreetMap mapnik style, in CartoCSS
+
+long_description    These are the CartoCSS map stylesheets for the Standard \
+                    map layer on OpenStreetMap.org.
+
+checksums           rmd160  f3effe425ea8c0d54f5437f22fe0181bb1005ea3 \
+                    sha256  be63d4fc06f1db447f1a249515f9fc83449f7f3027469d1cc051150ded3f85e9 \
+                    size    2719803
+
+python.default_version 39
+
+depends_lib-append  port:py${python.version}-psycopg2 \
+                    port:py${python.version}-requests \
+                    port:py${python.version}-yaml
+
+set datadir         ${prefix}/var/lib/${name}
+set docdir          ${prefix}/share/doc/${name}
+set configdir       ${prefix}/etc/${name}
+
+use_configure       no
+
+# The `scripts-get-external-data-py.diff` patch has been merged
+# into the upstream master at
+# https://github.com/gravitystorm/openstreetmap-carto/commit/4ec2dc9391c411e124c78b3ba1aad9173fea20cb
+# so it should be possible to remove it after the next release (> v5.3.1).
+patchfiles          style-fonts-mss.diff \
+                    scripts-get-external-data-py.diff
+
+variant postgresql12 conflicts postgresql13 description {Use with PostgreSQL 12} {
+    require_active_variants py${python.version}-psycopg2 postgresql12
+    depends_lib-append port:postgresql12
+}
+
+variant postgresql13 conflicts postgresql12 description {Use with PostgreSQL 13} {
+    require_active_variants py${python.version}-psycopg2 postgresql13
+    depends_lib-append port:postgresql13
+}
+
+if {![variant_isset postgresql12] && ![variant_isset postgresql13]} {
+    default_variants +postgresql13
+}
+
+post-patch {
+    reinplace "s#/usr/bin/env python3#${python.bin}#g" \
+        ${worksrcpath}/scripts/get-external-data.py \
+        ${worksrcpath}/scripts/generate_road_colours.py \
+        ${worksrcpath}/scripts/generate_shields.py \
+        ${worksrcpath}/scripts/indexes.py
+}
+
+build {}
+
+pre-destroot {
+    reinplace "s#data_dir: data#data_dir: ${datadir}#" ${worksrcpath}/external-data.yml
+}
+
+destroot {
+    xinstall -d ${destroot}${configdir}
+    xinstall -m 0755 -o nobody -g admin -d ${destroot}${datadir}
+    file copy ${worksrcpath} ${destroot}${prefix}/share/${name}
+}
+
+post-destroot {
+    file delete -force ${destroot}${prefix}/share/${name}/.github
+    file delete \
+        ${destroot}${prefix}/share/${name}/.dockerignore \
+        ${destroot}${prefix}/share/${name}/.editorconfig \
+        ${destroot}${prefix}/share/${name}/.gitattributes \
+        ${destroot}${prefix}/share/${name}/.github \
+        ${destroot}${prefix}/share/${name}/.gitignore \
+        ${destroot}${prefix}/share/${name}/.travis.yml \
+        ${destroot}${prefix}/share/${name}/Dockerfile \
+        ${destroot}${prefix}/share/${name}/Dockerfile.db \
+        ${destroot}${prefix}/share/${name}/Dockerfile.import \
+        ${destroot}${prefix}/share/${name}/docker-compose.yml
```

Comment:
this delete statement can be rewritten as:
```
     file delete {*}[glob -directory ${destroot}${prefix}/share/${name} Dockerfile* docker-compose.yml]

     foreach fn [glob -types hidden ${destroot}${prefix}/share/${name}/*] {
         if {[file tail ${fn}] ni {. ..}} {
             file delete ${fn}
         }
     }
```

Reply:
Thank you.  Those changes clean it up and are more future-proof too.